### PR TITLE
Makefile improvements

### PR DESCRIPTION
**Pull request by nanoant**
Long story short - compile time improvement, automatic build of CSS & bindata reducing the risk of forgetting to recompile them. See the commits for details.

### CODE REVIEW

**Comment by Unknwon** [comment]
No, generated content should always be part of main repo.

**Comment by tboerger** [comment]
I already did a massive makefile refactoring within the droneci pr :)

**Comment by nanoant** [comment]
@Unknwon okay, updated, this time without last commit removing bindata from repo

File: scripts/less.sh [hunk 1]
```diff
@@ -1,4 +0,0 @@
-#!/bin/sh
```
Comment (Unknwon) [inline scripts/less.sh]: Why remove this?

File: Makefile [hunk 2]
```diff
@@ -25,11 +29,18 @@ pack:
...
-	go-bindata -o=modules/bindata/bindata.go -ignore="\\.DS_Store|README.md" -pkg=bindata conf/...
+bindata: modules/bindata/bindata.go
+
+modules/bindata/bindata.go: $(DATA_FILES)
+	go-bindata -o=$@ -ignore="\\.DS_Store|README.md" -pkg=bindata conf/...
+
+less: public/css/gogs.css
+
+public/css/gogs.css: $(LESS_FILES)
+	lessc $< $@
```
Comment (Unknwon) [inline Makefile]: Does this really work without giving `../public/less/gogs.less`? .. just to confirm.

File: Makefile [hunk 3]
```diff
@@ -25,11 +29,18 @@ pack:
 
 release: build pack
 
-bindata: 
-	go-bindata -o=modules/bindata/bindata.go -ignore="\\.DS_Store|README.md" -pkg=bindata conf/...
+bindata: modules/bindata/bindata.go
+
+modules/bindata/bindata.go: $(DATA_FILES)
```
Comment (Unknwon) [inline Makefile]: Why add this command?

File: Makefile (hunk 2 above)
Comment (nanoant) [inline Makefile]: First element of `$(LESS_FILES)` is `public/less/gogs.less` so it doesn't need to be given explicitly.

File: Makefile [hunk 4]
```diff
@@ -1,17 +1,21 @@
 LDFLAGS += -X "github.com/gogits/gogs/modules/setting.BuildTime=$(shell date -u '+%Y-%m-%d %I:%M:%S %Z')"
 LDFLAGS += -X "github.com/gogits/gogs/modules/setting.BuildGitHash=$(shell git rev-parse HEAD)"
 
+DATA_FILES := $(shell find conf | sed 's/ /\\ /g')
+LESS_FILES := $(wildcard public/less/gogs.less public/less/_*.less)
```
Comment (Unknwon) [inline Makefile]: gogs.less already imports other .less, is `public/less/_*.less` part really needed?

File: Makefile (hunk 2 above)
Comment (Unknwon) [inline Makefile]: Thanks, I updated my original comment.

Comment (nanoant) [inline Makefile]: > Why use two commands other than combine to single one?

1st one is just an convenience alias to use `make less` via command line. 2nd is doing actual work and necessary to track dependencies and trigger rebuild.

File: Makefile [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
+GENERATED  := modules/bindata/bindata.go public/css/gogs.css
```
Comment (Unknwon) [inline Makefile]: Is this needed?

File: Makefile (hunk 2 above)
Comment (Unknwon) [inline Makefile]: How does it trigger rebuild?

**Comment by Unknwon** [comment]
I think all we need for this PR is the improvement of use `make less` to replace `less.sh`.

File: Makefile [hunk 6]
```diff
@@ -25,11 +29,18 @@ pack:
...
 release: build pack
 
-bindata: 
-	go-bindata -o=modules/bindata/bindata.go -ignore="\\.DS_Store|README.md" -pkg=bindata conf/...
+bindata: modules/bindata/bindata.go
+
+modules/bindata/bindata.go: $(DATA_FILES)
+	go-bindata -o=$@ -ignore="\\.DS_Store|README.md" -pkg=bindata conf/...
+
+less: public/css/gogs.css
```
Comment (Unknwon) [inline Makefile]: Why not just `lessc public/less/gogs.less public/css/gogs.css`?

File: Makefile (hunk 3 above)
Comment (Unknwon) [inline Makefile]: Same as below, why not just use one? Is this some high-tech on make command (very little knowledge on makefile)?

File: Makefile (hunk 5 above)
Comment (Unknwon) [inline Makefile]: Follow comments below, should this be changed to `bindata less`？

**Comment by Unknwon** [comment]
LGTM.

**Pull request merged**